app/models/post.rb

```diff
@@ -1,5 +1,14 @@
 class Post < ApplicationRecord
-  belongs_to :user, foreign_key: 'author_id'
-  has_many :likes, dependent: :destroy
-  has_many :comments, dependent: :destroy
+  belongs_to :author, class_name: 'User'
+  validates :author, presence: true
+  has_many :likes, inverse_of: 'post', dependent: :destroy
+  has_many :comments, inverse_of: 'post', dependent: :destroy
+
+  after_create :update_author_posts_counter
```

Comment:
- [ ] Good job using the Rails Active Callbacks, but don't you think it will be better to use the `after_save` instead, so that you only increase the counter when you are sure it has been saved.